hazelcast/src/test/java/com/hazelcast/spi/discovery/multicast/MemberToMemberDiscoveryTest.java

```diff
@@ -47,7 +48,9 @@ public void setUp() {
 
         System.setProperty(TestEnvironment.HAZELCAST_TEST_USE_NETWORK, "true");
         System.setProperty("java.net.preferIPv4Stack", "true");
-        factory = createHazelcastInstanceFactory(2);
+        if (factory == null) {
```

Comment:
Do we really need this? `factory` is already defined in HazelcastTestSupport so all cleanup, null checking etc. looks unnecessary to me. 

I believe we should remove `factory` variable in this class and depend on the one @ HazelcastTestSupport. Good cleanup opportunity for you 👍